py-rattler/rattler/version/with_source.py

```diff
@@ -0,0 +1,447 @@
+from __future__ import annotations
+
+from typing import List, Optional, Tuple, Union
+
+from rattler.version import Version
+
+from rattler.rattler import PyVersionWithSource, InvalidVersionError
+
+
+class VersionWithSource:
+    """
+    Holds a version and the string it was created from. This is useful if
+    you want to retain the original string the version was created from.
+    This might be useful in cases where you have multiple strings that
+    are represented by the same [`Version`] but you still want to be able to
+    distinguish them.
+
+    The string `1.0` and `1.01` represent the same version. When you print
+    the parsed version though it will come out as `1.0`. You loose the
```

Comment:
this can't be right (1.0 and 1.01 should not both print as `1.0`, right?)

Reply:
I think `1.1` and `1.01` are the same though.

Reply:
yes that's a mistake

Reply:
It's also incorrect in the rattler docs as well, need to fix that

Reply:
Great!